src/main/java/org/stellar/sdk/federation/FederationServer.java

```diff
@@ -76,39 +74,34 @@ public FederationServer(String serverUri, InternetDomainName domain) {
    * @throws StellarTomlNotFoundInvalidException Stellar.toml file was not found or was malformed.
    * @return FederationServer
    */
-  public static FederationServer createForDomain(InternetDomainName domain) {
-    Executor executor = Executor.newInstance(FederationServer.httpClient);
-    URI stellarTomlUri;
+  public static FederationServer createForDomain(String domain) {
+    StringBuilder uriBuilder = new StringBuilder();
+    uriBuilder.append(httpsConnection ? "https://" : "http://");
+    uriBuilder.append(domain);
+    uriBuilder.append("/.well-known/stellar.toml");
+    HttpUrl stellarTomlUri = HttpUrl.parse(uriBuilder.toString());
+    OkHttpClient httpClient = FederationServer.createHttpClient();
+
+    Request request = new Request.Builder().get().url(stellarTomlUri).build();
+    Response response;
     try {
-      StringBuilder uriBuilder = new StringBuilder();
-      uriBuilder.append("https://");
-      uriBuilder.append(domain.toString());
-      uriBuilder.append("/.well-known/stellar.toml");
-      stellarTomlUri = new URI(uriBuilder.toString());
-    } catch (URISyntaxException e) {
-      throw new RuntimeException(e);
+      response = httpClient.newCall(request).execute();
+    } catch (IOException e) {
+      throw new ConnectionErrorException();
+    }
+
+    if (response.code() >= 300) {
```

Comment:
ResponseBody will never be closed -> memory leak

Reply:
Check my previous comment. `response` is `Response` object, not `ResponseBody` object. And we don't have to close it because we're using `string()` method that closes it.

Reply:
Actually, you may be right. OkHttp docs are not super clear on if `Response` must be closed even if `ResponseBody` was not requested/processed.

Reply:
OK, it looks like we need to close the `Response` even if it's body is not used: https://github.com/square/okhttp/issues/2311. Will fix this, thanks for catching!